controller/common/src/Controller/Common/Order/Standard.php

```diff
@@ -347,7 +348,16 @@ protected function updateStock( \Aimeos\MShop\Order\Item\Iface $orderItem, int $
 
 				foreach( $items as $item )
 				{
-					$stockManager->decrease( [$item->getProductId() => $how * -1 * $item->getQuantity()], $item->getStockType() );
+					$prodId = $item->getProductId();
+
+					if( $item->getType() === 'select' ) {
+						$prodId = $productManager->get( $item->getProductId(), ['product' => ['default']] )
+							->getRefItems( 'product', null, 'default' )
+							->where( 'product.code', '==', $item->getProductCode() )
+							->getId()->first();
+					}
```

Comment:
There's already a switch statement below with calls `updateStockSelection()`. If you move your code to that method, testing may be easier.